packages/test/src/browser/view/test-execution-progress-service.ts

```diff
@@ -0,0 +1,54 @@
+// *****************************************************************************
+// Copyright (C) 2024 STMicroelectronics and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { inject, injectable } from '@theia/core/shared/inversify';
```

Comment:
Not sure why we need this class. Why isn't this handled in `TestService`?

Reply:
Or alternatively, don't go through the test `TestService` but notify this service directly from `test-main.ts`.

Reply:
I had an issue with the architecture there: the test views depend on the TestService, so I could not have the TestService to depend on them.
But yes, I can notify it from test-main.ts. I will propose a new version with TestExecutionProgressService directly used in TestingMain.

Reply:
The new commit has shifted a bit the way it is architectured, as you suggested. TestingMain is directly invoking the progress service and not channeling events through main test service anymore.